File: Data Structures & Algorithms/merge-intervals/submission-6.java
class Solution {
    public int[][] merge(int[][] intervals) {
        List<int[]> result = new ArrayList<>();
        Arrays.sort(intervals, (x,y) -> Integer.compare(x[0],y[0]));
        int[] current = intervals[0];

        for(int i = 1; i < intervals.length; ++i){
            int[] next = intervals[i];
            if(current[1] >= next[0]){
                current[1] = Math.max(next[1], current[1]);
            }else{
                result.add(current);
                current = next;
            }
        }
        result.add(current);
        return result.toArray(new int[result.size()][2]);
    }
}

File: Data Structures & Algorithms/insert-new-interval/submission-6.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        intervals.append(newInterval)
        intervals.sort()
        result = []
        curr = intervals[0]
        for nxt in intervals[1:]:
            if nxt[0] <= curr[1]:
                curr[0] = min(curr[0], nxt[0])
                curr[1] = max(curr[1], nxt[1])
            else:
                result.append(curr)
                curr = nxt
        result.append(curr)
        return result



